malcolm/modules/scanning/controllers/runnablecontroller.py

```diff
@@ -50,9 +50,12 @@ def get_steps_per_run(generator, axes_to_move):
                 "Axis %s is not in %s" % (axis, axes_to_move)
             axes_set.remove(axis)
         # Now multiply by the dimensions to get the number of steps
-        steps *= dim.size
-    return steps
+        steps[0] *= dim.size
+
+    if len(breakpoints) > 1:
+        steps = breakpoints
```

Comment:
We should validate that the sum of the breakpoints is less than or equal to steps[0]